### PR TITLE
feat(consumer-group describe): show unconsumed partitions

### LINKED ISSUE #778: Show unconsumed partitions for a consumer group
### Feature or problem description

The `consumergroup describe` command lists all members of a consumer group. Users will want to see if there are unconsumed partitons.

### Proposed solution

If a consumer's "memberId" is null, this means that it is unconsumed. we can use this to calculate the number of unconsumed partitions and display it as part of the `describe` command.

### PR DESCRIPTION
**Pull request by rkpattnaik780**
User should be able to see the number of unconsumed partitions in a consumer group using the `rhoas kafka consumer-group describe` command.

Closes #778 

Screenshot from 2021-07-22 16-29-54

### Verification Steps

1. Start the mock server by running `make mock-api/start`.
2. Login
```
./rhoas login --api-gateway=http://localhost:8000
```
3. Create a kafka instance
```
./rhoas kafka create <name-of-instance>
```
4. Select the kafka instance running the command `./rhoas kafka use --id <id-of-instance>`.
```
./rhoas kafka use --id <id-of-instance>
```
5. View the consumer group with id `consumer_group_1`.
```
./rhoas kafka consumer-group describe --id consumer_group_1
```

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### Checklist

- [ ] Documentation added for the feature
- [X] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

**Comment by rkpattnaik780** [comment]
To test this PR against mock, this [fix](https://github.com/redhat-developer/app-services-sdk-js/pull/123) needs to be merged.

**Comment by craicoverflow** [comment]
Is this ready to be reviewed?

File: pkg/cmd/kafka/consumergroup/describe/describe.go [hunk 1]
```diff
@@ -166,6 +166,11 @@ func mapConsumerGroupDescribeToTableFormat(consumers []kafkainstanceclient.Consu
 			CurrentOffset: int(consumer.GetOffset()),
 			OffsetLag:     int(consumer.GetLag()),
 		}
+
+		if consumer.GetMemberId() == "" {
+			row.MemberID = "unconsumed"
```
Comment (craicoverflow) [inline pkg/cmd/kafka/consumergroup/describe/describe.go:171]: When displayed in the table it does not stand out enough - users might think this is what the actual member ID is.

I think we should do something to highlight it, perhaps by changing the color of the text and capitalising it. Or something along those lines?

**Review by craicoverflow — CHANGES_REQUESTED** [review]

File: pkg/cmd/kafka/consumergroup/describe/describe.go (hunk 1 above)
Comment (rkpattnaik780) [reply pkg/cmd/kafka/consumergroup/describe/describe.go:171]: Is it possible to italicize it?

Comment (craicoverflow) [reply pkg/cmd/kafka/consumergroup/describe/describe.go:171]: Yes it is. I suggest play around with some different formats and see what fits the best.

Comment (rkpattnaik780) [reply pkg/cmd/kafka/consumergroup/describe/describe.go:171]: Screenshot from 2021-07-28 13-42-14
Screenshot from 2021-07-28 13-42-37
Screenshot from 2021-07-28 13-43-04
Screenshot from 2021-07-28 13-44-03
Screenshot from 2021-07-28 13-56-49

Tried a few variations, italics don't seem to be very much distinct. I guess bold and `info` will be better, wdyt?

Comment (craicoverflow) [reply pkg/cmd/kafka/consumergroup/describe/describe.go:171]: Upon seeing these colours I don't think any of them suit in this setting. How about yellow?

image

Comment (craicoverflow) [reply pkg/cmd/kafka/consumergroup/describe/describe.go:171]: How does the UI represent this?

Comment (rkpattnaik780) [reply pkg/cmd/kafka/consumergroup/describe/describe.go:171]: UI is supposed to represent it as small-cased italics. 
_unconsumed_

**Review by craicoverflow — APPROVED** [review]

**Comment by rkpattnaik780** [comment]
Here's the final preview.

Screenshot from 2021-07-28 18-56-39

File: pkg/cmd/kafka/consumergroup/describe/describe.go (hunk 1 above)
Comment (rkpattnaik780) [reply pkg/cmd/kafka/consumergroup/describe/describe.go]: UI doesn't display a count of unconsumed partitions with a sub-heading though. I feel it should be there, so I stuck to that in CLI. Wdyt?

**Comment by craicoverflow** [comment]
Looks good! Happy for it to be merged

File: pkg/cmd/kafka/consumergroup/describe/describe.go (hunk 1 above)
Comment (craicoverflow) [reply pkg/cmd/kafka/consumergroup/describe/describe.go]: Yeah I think it is okay to have it in the CLI as the UI and CLI do not need to exactly match - just as long as the minimum required data is showed.

**Pull request merged**